lib/core/src/chain/bitcoin.rs

```diff
@@ -58,8 +60,38 @@ impl ElectrumClient {
 
         Ok(Self { client, tip })
     }
+
+    async fn get_script_history_with_retry(
+        &self,
+        script: &Script,
+        retries: u64,
+    ) -> Result<Vec<History>> {
+        let script_hash = sha256::Hash::hash(script.as_bytes())
+            .to_byte_array()
+            .to_hex();
+        info!("Fetching script history for {}", script_hash);
+        let mut script_history = vec![];
+
+        let mut retry = 1;
+        while retry < retries {
```

Comment:
Its easy to get into an endless loop if `retries` is 0.

Maybe go for
```rust
let mut retry = 0;
while retry < retries { ... }
```

Reply:
My mistake, its not an endless loop, its an edge case that would cause it to not fetch the history at all.

Reply:
You're right, fixed